python/test/kv/kvBench.py

```diff
@@ -0,0 +1,183 @@
+    vlst = [np.arange(25, dtype=np.uint32), np.arange(10000, dtype=np.uint32), \
+    np.arange(100000, dtype=np.uint32), np.arange(25000000, dtype=np.uint32)] # np.random is slow; np.range and np.shuffle
+    assert len(klst) == len(vlst)
+    if is_mixed:
+        print("Test with mixed sizes of matrices")
+        print("Test Direct")
+        kv = ff.kv.Local()
+        time_list = []
+        for j in range(11):
+            time_once = 0
+            for i in range(len(klst)):
+                k, v = klst[i], vlst[i]
+                start_time = time.time()
+                testDirect(kv, k, v)
+                end_time = time.time()
+                time_once += end_time - start_time
+            if j == 0 and not is_cold:
+                continue
+            time_list.append(time_once)
+        time_list = np.array(time_list)
+        print("Whole time list is:")
+        print(time_list)
+        print("Mean is: " + str(np.mean(time_list)))
+        print("Variance is: " + str(np.var(time_list)))
+        print("-"*30)
+        print("Test Redis")
+        kv = ff.kv.Redis(pwd=ff.redisPwd, serialize=True)
+        time_list = []
+        for j in range(11):
+            time_once = 0
+            for i in range(len(klst)):
+                k, v = klst[i], vlst[i]
+                start_time = time.time()
+                testRedis(kv, k, v)
+                end_time = time.time()
+                time_once += end_time - start_time
+            if j == 0 and not is_cold:
+                continue
+            time_list.append(time_once)
+        time_list = np.array(time_list)
+        print("Whole time list is:")
+        print(time_list)
+        print("Mean is: " + str(np.mean(time_list)))
+        print("Variance is: " + str(np.var(time_list)))
+        print("-"*30)
+        print("Test Anna")
+        kv = ff.kv.Anna("127.0.0.1", "127.0.0.1", local=True)
+        time_list = []
+        for j in range(11):
+            time_once = 0
+            for i in range(len(klst)):
+                k, v = klst[i], vlst[i]
+                start_time = time.time()
+                testAnna(kv, k, v)
+                end_time = time.time()
+                time_once += end_time - start_time
+            if j == 0 and not is_cold:
+                continue
+            time_list.append(time_once)
+        time_list = np.array(time_list)
+        print("Whole time list is:")
+        print(time_list)
+        print("Mean is: " + str(np.mean(time_list)))
+        print("Variance is: " + str(np.var(time_list)))
+        print("-"*30)
+    else:
+        for i in range(len(klst)):
+            k, v = klst[i], vlst[i]
+            print("Test for " + k)
+            if is_manyRW:
+                print("Test with mutiple reads and writes")
+            print("Test Direct")
+            kv = ff.kv.Local()    
+            time_list = []
+            for j in range(11):
+                if is_manyRW:
```

Comment:
You might want to put in a short sleep (0.5s or something) between iterations for manyRW to let the system quiesce. Basically, the single RW is still happen in a quick burst (that 'j in range(11)' line), the manyRW just does a few more iterations (basically it does 55 iterations instead of 11) but it basically measures the same thing. A short sleep between iterations lets the system settle down again.